app/assets/javascripts/models/beta/projectBoard.js

```diff
@@ -51,14 +52,84 @@ export const translateWord = (operand, word, translations) =>
     ? word
     : I18n.t(`story.${operand}.${translations[word] || word}`, { locale: I18n.defaultLocale });
 
-const translatedOperands = ['type','state'];
+const translatedOperands = ['type', 'state'];
 
 export const haveTranslation = operand => translatedOperands.includes(operand)
 
 export const isEnglishLocale = () => currentLocale() === 'en';
 
 const currentLocale = () => I18n.currentLocale();
 
+// Drag And drop utils
+
+const calculatePosition = (aboveStory, bellowStory) => {
+  if (bellowStory === undefined) return Number(aboveStory.position) + 1;
+  if (aboveStory === undefined) return Number(bellowStory.position) - 1;
+  return (Number(bellowStory.position) + Number(aboveStory.position)) / 2;
+};
+
+export const getNewPosition = (
+  destinationIndex,
+  sourceIndex,
+  storiesArray,
+  isSameColumn,
+  storyType,
+) => {
+  if(storiesArray.length === 0){
+    return 1; // if array is empty than set 1 to story position
+  }
+
+  if (!isSameColumn || sourceIndex > destinationIndex) {
+    return calculatePosition(
+      storiesArray[destinationIndex - 1],
+      storiesArray[destinationIndex],
+    );
+  }
+  
+  return calculatePosition(
+    storiesArray[destinationIndex],
+    storiesArray[destinationIndex + 1],
+  );
+};
+
+// reorder the array
+export const moveTask = (
+  sourceArray,
+  destinationArray,
+  sourceIndex,
+  destinationIndex,
+) => {
+  const newSourceArray = sourceArray;
+  const [removed] = newSourceArray.splice(sourceIndex, 1);
+  const newDestinationArray = destinationArray;
+  newDestinationArray.splice(destinationIndex, 0, removed);
+  return [...newDestinationArray];
+};
+
+export const getNewSprints = (newStories, sprints, sprintIndex) =>
+  sprints.map((sprint, index) =>
+    index === sprintIndex ? { ...sprint, stories: newStories } : sprint,
+  );
+
+export const getNewState = (isSameColumn, dropColumn, currentState) => {
+  if(isSameColumn) {
+    return currentState;
+  }
+  return dropColumn === 'chillyBin' ? status.UNSCHEDULED : status.UNSTARTED;
+}
+
+export const getBacklogStories = (sprints, index) => sprints.length === 0 ? [{stories: []}] : sprints[index].stories;
```

Comment:
You can use [`isEmpty`](https://underscorejs.org/#isEmpty) from `underscore` here